tests/test_output.py

```diff
@@ -131,47 +155,84 @@ def test_message_copy_with_inline_text(
 
     # copy
     if interactive:
-        tui = spawn(COPIER_PATH + ("copy", str(src), str(dst)), timeout=10)
+        tui = spawn(COPIER_PATH + ("copy", "-r", "v1", str(src), str(dst)), timeout=10)
         expect_prompt(tui, "project_name", "str")
         tui.sendline("myproj")
         tui.expect_exact(pexpect.EOF)
     else:
-        run_copy(str(src), dst, data={"project_name": "myproj"})
+        run_copy(str(src), dst, data={"project_name": "myproj"}, vcs_ref="v1")
 
+    assert (dst / "version.txt").read_text() == "v1"
     _, err = capsys.readouterr()
-    assert re.search(pattern.format(project_name="myproj"), err, flags=re.S)
+    assert re.search(copy_pattern.format(project_name="myproj"), err, flags=re.S)
 
     # recopy
     if interactive:
-        tui = spawn(COPIER_PATH + ("recopy", str(dst)), timeout=10)
+        tui = spawn(COPIER_PATH + ("recopy", "-r", "v1", str(dst)), timeout=10)
         expect_prompt(tui, "project_name", "str")
         tui.sendline("_new")
         tui.expect_exact(pexpect.EOF)
     else:
-        run_recopy(dst, data={"project_name": "myproj_new"})
+        run_recopy(dst, data={"project_name": "myproj_new"}, vcs_ref="v1")
 
+    assert (dst / "version.txt").read_text() == "v1"
     _, err = capsys.readouterr()
-    assert re.search(pattern.format(project_name="myproj_new"), err, flags=re.S)
+    assert re.search(copy_pattern.format(project_name="myproj_new"), err, flags=re.S)
+
+    with local.cwd(dst):
+        git("init")
+        git("add", ".")
+        git("commit", "-m1")
+
+    # clear capture output log
+    capsys.readouterr()
+
+    # update
+    if interactive:
+        tui = spawn(COPIER_PATH + ("update", str(dst)), timeout=10)
+        expect_prompt(tui, "project_name", "str")
+        tui.sendline("_update")
+        tui.expect_exact(pexpect.EOF)
+    else:
+        run_update(dst, data={"project_name": "myproj_new_update"}, overwrite=True)
+
+    assert (dst / "version.txt").read_text() == "v2"
+    _, err = capsys.readouterr()
+    assert re.search(
+        (
+            r"^"
+            r"Updating on (linux|macos|windows)"
+            r".+"
+            r"Project {project_name} successfully updated"
+            r"\s*"
+            r"$"
```

Comment:
https://docs.python.org/3/library/re.html#re.VERBOSE

It's beautier to use a verbose regexp IMHO.

Reply:
I'd do it slightly different. I think it makes the code clearer TBH:

```py
assert re.search(
    fr"""
    ^Updating\ on\ (linux|macos|windows).+
    Project\ {project_name}\ successfully\ updated\s*$
    """,
    err,
    flags=re.S | re.X,
)
```

Reply:
Thanks, that looks more concise. :+1: I've added a commit that updates the regexes using verbose syntax also for copy/recopy/update messages. So this PR should probably be rebase-merged, not squash-merged.